yjit_codegen.c

```diff
@@ -3511,6 +3512,47 @@ gen_tostring(jitstate_t* jit, ctx_t* ctx)
     return YJIT_KEEP_COMPILING;
 }
 
+static codegen_status_t
+gen_toregexp(jitstate_t* jit, ctx_t* ctx)
+{
+    rb_num_t opt = jit_get_arg(jit, 0);
+    rb_num_t cnt = jit_get_arg(jit, 1);
+
+    // Save the PC and SP because we might make a Ruby call for
+    // Kernel#set_trace_var
+    jit_save_pc(jit, REG0);
+    jit_save_sp(jit, ctx);
+
+    x86opnd_t values_ptr = ctx_sp_opnd(ctx, -(sizeof(VALUE) * (uint32_t)cnt));
+    ctx_stack_pop(ctx, cnt);
+
+    mov(cb, C_ARG_REGS[0], imm_opnd(0));
+    mov(cb, C_ARG_REGS[1], imm_opnd(cnt));
+    lea(cb, C_ARG_REGS[2], values_ptr);
+    call_ptr(cb, REG0, (void *)&rb_ary_tmp_new_from_values);
+
+    // Save the array so we can clear it later
+    push(cb, RAX);
+    push(cb, RAX); // Alignment
+    mov(cb, C_ARG_REGS[0], RAX);
+    mov(cb, C_ARG_REGS[1], imm_opnd(opt));
+    call_ptr(cb, REG0, (void *)&rb_reg_new_ary);
+
+    // The actual regex is in RAX now.  Pop the temp array from
+    // rb_ary_tmp_new_from_values into C arg regs so we can clear it
+    pop(cb, REG1); // Alignment
+    pop(cb, C_ARG_REGS[0]);
+
+    // The value we want to push on the stack is in RAX right now
+    x86opnd_t stack_ret = ctx_stack_push(ctx, TYPE_UNKNOWN);
+    mov(cb, stack_ret, RAX);
+
+    // Clear the temp array. FIXME: is this necessary?
+    call_ptr(cb, REG0, (void *)&rb_ary_clear);
```

Comment:
Is it necessary? 😅

Reply:
Yea - we don't know if this is necessary in the original (I noted that in the commit message)